pennylane/templates/subroutines/qsvt.py

```diff
@@ -464,3 +466,217 @@ def _qsp_to_qsvt(angles):
     update_vals = qml.math.convert_like(update_vals, angles)
 
     return angles + update_vals
+
+
+def _complementary_poly(P):
+    r"""
+    Computes the complementary polynomial Q given a polynomial P.
+
+    The polynomial Q is complementary of P if satisfies the following equation:
+
+    .. math:
+
+        |P(e^{i\theta})|^2 + |Q(e^{i\theta})|^2 = 1, \quad \forall \theta \in \left[0, 2\pi\right]
+
+    The method is based on computing an auxiliary polynomial R, finding its roots,
+    and reconstructing Q by using information extracted of those roots.
+    For more details see reference `arXiv:2308.01501 <https://arxiv.org/abs/2308.01501>`_.
+
+    Args:
+        P (array-like): Coefficients of the complex polynomial P.
+
+    Returns:
+        array-like: Coefficients of the complementary polynomial Q.
+    """
+    poly_degree = len(P) - 1
+
+    # Build the polynomial R(z) = z^degree * (1 - conj(P(1/z)) * P(z)), deduced from (eq.33) and (eq.34)
+    R = Polynomial.basis(poly_degree) - Polynomial(P) * Polynomial(np.conj(P[::-1]))
```

Comment:
Yep, actually this is a bit unintuitive at first glance

 The key here is that if you have $P(z) = a + bz^1 + cz^2$, then
$P(1/z) = a + bz^{-1} + cz^{-2}$
Which is equal to 
$P(1/z)= z^{-2} (a z^2 + bz + c) $
Which is equal to
$P(1/z) = z^{-2}(P[::-1])$

and it can be simplified as the expression showed